src/views/components/item.jsx

```diff
@@ -1,16 +1,57 @@
 import React from 'react';
-
+import ItemStore from '../../store/item-store.js';
+import { Link } from 'react-router';
+import { searchItem } from '../../lib/api-client.js';
 export default class Item extends React.Component {
+    constructor(props) {
+        super(props);
+        if (props.item === undefined)
+            this.state = {item: null};
+        else
+            this.state = {item: props.item};
+    }
 
-    render() {
-        if (this.props.item === null) {
-            return <i>No item found</i>;
+    componentDidMount() {
+        if (this.state.item === null) {
+            searchItem(this.props.params.address).then(() => {
+                this.setState({
+                    item: ItemStore.getItem()
+                });
+            });
```

Comment:
Ahhh i see the confusion. It indeed would not work and is actually un-testable till someone puts in the ability to view the items of a model. See: https://github.com/TheFourFifths/consus-client/blob/view-all-items-tff35/src/views/components/item.jsx. Nice catch, i forgot it.